examples/player/simid_player.js

```diff
@@ -609,16 +602,78 @@ class SimidPlayer {
     console.log('The creative has asked for the player to ping ' + requestedUrlArray);
   }
 
+  /**
+   * Called when video playback is complete.
+   * @private
+   */
+  videoComplete() {
+    this.simidProtocol.sendMessage(MediaMessage.ENDED);
+
+    if (this.requestedDuration_ == NO_REQUESTED_DURATION) {
+      this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
+    }
+  }
+
+  /**
+   * Called when creative requests a change in duration of ad.
+   * @private
+   */
   onRequestChangeAdDuration(incomingMessage) {
-    if (this.requestedDuration_ != NO_REQUESTED_DURATION) {
-      // TODO: Support multiple change duration requests.
-      this.simidProtocol.reject(incomingMessage);
+    const originalDuration = this.requestedDuration_;
+    this.requestedDuration_  = incomingMessage.args['duration'];
+    if (this.requestedDuration_ != UNLIMITED_DURATION && this.requestedDuration_< 0) {
+      const durationErrorMessage = {
+        errorCode: PlayerErrorCode.UNSUPPORTED_TIME,
+        message: 'A negative duration is not valid.'
+      }
+      this.requestedDuration_ = originalDuration;
+      this.simidProtocol.reject(incomingMessage, durationErrorMessage);
+    } else if (this.requestedDuration_ == UNLIMITED_DURATION) {
+      //Note: Users can end the ad using the close ad button on the player
+      return;
+    } else {
+      //If requested duration is any other acceptable value
+      this.changeAdEndTime();
+      this.simidProtocol.resolve(incomingMessage);
     }
-    const requestedDuration  = incomingMessage.args['duration'];
-    this.requestedDuration_ = requestedDuration;
-    this.simidProtocol.resolve(incomingMessage);
   }
 
+  /**
+   * Compares the requested change duration with the current ad time
+   *  to determine when to stop the ad, every 250 milliseconds.
+   * @private
+   */
+  changeAdEndTime() {
+    this.interval_ = setInterval(() => {
```

Comment:
Make sure to add `this.interval_` to the constructor (even if you're just setting it to null), and include type annotations

Reply:
Added!